src/Test/Hspec/Expectations/Contrib.hs

```diff
@@ -24,3 +30,36 @@ isRight :: Either a b -> Bool
 isRight (Left  _) = False
 isRight (Right _) = True
 #endif
+
+-- |
+-- If you have a test case that has multiple assertions, you can use the
+-- 'annotate' function to provide a string message that will be attached to
+-- the 'Expectation'.
+--
+-- @
+-- describe "annotate" $ do
+--   it "adds the message" $ do
+--     annotate "obvious falsehood" $ do
+--       True `shouldBe` False
+--
+-- ========>
+--
+-- 1) annotate, adds the message
+--       obvious falsehood
+--       expected: False
+--        but got: True
+-- @
+--
+-- @since 0.8.3
+annotate :: (HasCallStack) => String -> Expectation -> Expectation
```

Comment:
@parsonsmatt for what reason do we need the `HasCallStack` constraint here?

Reply:
Great question. I don't recall why I put that there. It builds fine without it. All of the other functions in `Test.Hspec.Expectations` have it, so I suppose I was following the pattern. I've removed it.